### PR TITLE
47 refactor logging

### LINKED ISSUE #47: lib/logging.rb - rewrite nested methods
Rubocop doesn't like our nested methods, and [neither should we](https://github.com/bbatsov/ruby-style-guide#no-nested-methods).

```
lib/logging.rb:16:7: W: Method definitions must not be nested. Use lambda instead.
      def logger 
      ^^^^^^^
```

### PR DESCRIPTION
**Pull request by tamacodechi**
Closes #47 

I'm a little unsure about this one because Rubocop requests a lambda instead of nested definitions we have. All is good when I define it using a lambda but rubocop doesn't like the "unused" declaration ``` logger = -> ``` so I call it directly underneath, but surely we're calling this logger somewhere else which is why it was defined as it was before, right?

### CODE REVIEW

**Comment by clemenshelm** [comment]
I'm almost certain that this code doesn't work. What's the exact error that Rubocop gives you?

**Comment by tamacodechi** [comment]
```
lib/logging.rb:16:7: W: Method definitions must not be nested. Use lambda instead.
      def logger ...
      ^^^^^^^^^^
```

**Comment by tamacodechi** [comment]
To be honest I have no idea how our logger works. I will definitely be reading up on the Logger class this week in my free time, but if I'm still lost, would it be okay to spend like 10 mins pairing just going through how we've implemented it?

File: lib/logging.rb [hunk 1]
```diff
@@ -13,9 +13,8 @@ def logger
   # Addition
   def self.included(base)
     class << base
-      def logger
-        Logging.logger
-      end
```
Comment (clemenshelm) [inline lib/logging.rb:18]: Maybe changing the method definition to

```ruby
define_method(:logger) { Logging.logger }
```

will fix the Rubocop issue.

Comment (tamacodechi) [reply lib/logging.rb:18]: Yes this is so much better and works beautifully. Thank you!! 👏  🎉

**Review by clemenshelm — APPROVED** [review]
Please resolve the conflicts, then we can merge it.

**Pull request merged**